ansible-ipi-install/roles/node-prep/tasks/10_validation.yml

```diff
@@ -1,3 +1,38 @@
+---
+- name: Gather facts for python version on localhost
+  setup:
+    gather_subset:
+      - facter
+  delegate_to: localhost
+  tags: validation
+
+- name:  Gather the package facts on localhost
+  package_facts:
+    manager: auto
+  delegate_to: localhost
+  tags: validation
+
+- debug:
+    msg: "{{ ansible_python_version }}"
+    verbosity: 2
+  tags: validation
+
+- fail: msg="The playbook requires python2-dns and python2-netaddr, install python2-dns python2-netaddr on your localhost"
+  when: 
+    - ansible_python_version is version('3.0.0', '<')
+    - ('python2-dns' not in ansible_facts.packages or 'python2-netaddr' not in ansible_facts.packages)
+  tags: validation
+
+- fail: msg="The playbook requires python3-dns and python3-netaddr, install python3-dns python3-netaddr on your localhost"
+  when:
+    - ansible_python_version is version('3.0.0', '>=')
+    - ('python3-dns' not in ansible_facts.packages or 'python3-netaddr' not in ansible_facts.packages)
```

Comment:
Can you use with_items to avoid this list? Also, if more packages are needed, you just need to add them to the list